coredns/plugin/record.go

```diff
@@ -102,10 +102,20 @@ func (lh *Lighthouse) getClusterIPForSvc(pReq *recordRequest) (*serviceimport.DN
 	record, found, isLocal := lh.ServiceImports.GetIP(pReq.namespace, pReq.service, pReq.cluster, localClusterID, lh.ClusterStatus.IsConnected,
 		lh.EndpointsStatus.IsHealthy)
 
+	var serviceImportIP string
+	if record != nil && found {
+		serviceImportIP = record.IP
+	}
+
 	getLocal := isLocal || pReq.cluster != "" && pReq.cluster == localClusterID
 	if found && getLocal {
 		record, found = lh.LocalServices.GetIP(pReq.service, pReq.namespace)
 	}
 
+	// for local request with empty cluster in request prefer the IP retrieved from serviceimport
+	if getLocal && pReq.cluster == "" && serviceImportIP != "" && record != nil {
+		record.IP = serviceImportIP
+	}
+
```

Comment:
I"m not sure I follow the logic with the changes. But I think the `LocalServices` code was added for globalnet b/c the global IPs were locally routable, in which case we can remove it and none of L105-118 should be needed.  @vthapar @aswinsuryan

Reply:
> I"m not sure I follow the logic with the changes. But I think the `LocalServices` code was added for globalnet b/c the global IPs weren't locally routable, in which case we can remove it and none of L105-118 should be needed. @vthapar @aswinsuryan

Also thought like you at first, and that was also the first version of this PR, but all SRV and HEADLESS SRV tests failed.
Let's see what @aswinsuryan @vthapar think

Reply:
Hmm...we should investigate why the SRV tests fail in that case - perhaps there's assumptions/checks in the tests that need to be adjusted.  Maybe it's just me but I find L105-118 a bit confusing and hard to follow - I think it would be ideal if we could at least simplify this section if not remove it altogether.

Reply:
It could be a test issue, if e2e is not failing and the use case is working when we removed those lines.  

if not, we do retrieve the port information as well in GetIP method and it could be a case where we depend on this behavior.

Reply:
I tried it w/o the local service lookup and all the E2E tests pass but the headless service plugin handler tests fail. This is an issue with the test setup - it turns out they're  not actually testing with a headless service b/c the general test setup adds a non-headless service first. I pushed [this PR](https://github.com/submariner-io/lighthouse/pull/1068) to fix it. With that we can remove the `LocalServices` functionality.

Reply:
Ack, 10x